tools/inx-indexer/main.go

```diff
@@ -0,0 +1,278 @@
+	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/keepalive"
+
+	indexerpkg "github.com/gohornet/hornet/pkg/indexer"
+	indexer_server "github.com/gohornet/hornet/pkg/indexer/server"
+	"github.com/gohornet/hornet/pkg/inx"
+	"github.com/gohornet/hornet/pkg/model/milestone"
+	"github.com/gohornet/hornet/pkg/model/utxo"
+	"github.com/gohornet/hornet/pkg/utils"
+	"github.com/iotaledger/hive.go/serializer/v2"
+	iotago "github.com/iotaledger/iota.go/v3"
+)
+
+const (
+	APIRoute = "indexer/v1"
+)
+
+func ConvertINXOutput(output *inx.LedgerOutput) (*utxo.Output, error) {
+	outputID := output.UnwrapOutputID()
+	messageID := output.UnwrapMessageID()
+	milestoneIndex := milestone.Index(output.GetMilestoneIndexBooked())
+	milestoneTimestamp := uint64(output.GetMilestoneTimestampBooked())
+	o, err := output.UnwrapOutput(serializer.DeSeriModePerformValidation)
+	if err != nil {
+		return nil, err
+	}
+
+	return utxo.CreateOutput(outputID, messageID, milestoneIndex, milestoneTimestamp, o), nil
+}
+
+func ConvertINXSpent(spent *inx.LedgerSpent) (*utxo.Spent, error) {
+	output, err := ConvertINXOutput(spent.GetOutput())
+	if err != nil {
+		return nil, err
+	}
+	targetTransactionID := spent.UnwrapTransactionIDSpent()
+	milestoneIndex := milestone.Index(spent.GetMilestoneIndexSpent())
+	milestoneTimestamp := uint64(spent.GetMilestoneTimestampSpent())
+
+	return utxo.NewSpent(output, targetTransactionID, milestoneIndex, milestoneTimestamp), nil
+}
+
+func ConvertINXOutputs(outputs []*inx.LedgerOutput) (utxo.Outputs, error) {
+	var out utxo.Outputs
+	for _, output := range outputs {
+		o, err := ConvertINXOutput(output)
+		if err != nil {
+			return nil, err
+		}
+		out = append(out, o)
+	}
+	return out, nil
+}
+
+func ConvertINXSpents(spents []*inx.LedgerSpent) (utxo.Spents, error) {
+	var sp utxo.Spents
+	for _, spent := range spents {
+		s, err := ConvertINXSpent(spent)
+		if err != nil {
+			return nil, err
+		}
+		sp = append(sp, s)
+	}
+	return sp, nil
+}
+
+func fillIndexer(client inx.INXClient, indexer *indexerpkg.Indexer) error {
+	importer := indexer.ImportTransaction()
+	stream, err := client.ReadUnspentOutputs(context.Background(), &inx.NoParams{})
+	if err != nil {
+		panic(err)
+	}
+	var ledgerIndex milestone.Index
+	for {
+		message, err := stream.Recv()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return err
+		}
+
+		ledgerOutput := message.GetOutput()
+		iotaOutput, err := ledgerOutput.UnwrapOutput(serializer.DeSeriModeNoValidation)
+		if err != nil {
+			return err
+		}
+		outputLedgerIndex := milestone.Index(message.GetLedgerIndex())
+		output := utxo.CreateOutput(ledgerOutput.UnwrapOutputID(), ledgerOutput.UnwrapMessageID(), milestone.Index(ledgerOutput.GetMilestoneIndexBooked()), uint64(ledgerOutput.GetMilestoneTimestampBooked()), iotaOutput)
+		if err := importer.AddOutput(output); err != nil {
+			return err
+		}
+		if outputLedgerIndex > ledgerIndex {
```

Comment:
I like when both is on the same side after the condition :innocent: 
```suggestion
		if ledgerIndex < outputLedgerIndex {
```